tests/hw-accel/amdgpu/basic/tests/basic-test.go

```diff
@@ -116,5 +117,41 @@ var _ = Describe("AMD GPU Basic Tests", Ordered, Label(amdparams.LabelSuite), fu
 						nodeLabellerEnabled, *deviceconfig.GetEnableNodeLabeller(deviceConfigBuilderNew)))
 			}
 		})
+
+		It("Device Plugin", func() {
+
+			podsBuilder, podsBuilderErr := pods.GetPodsFromNamespaceByPrefixWithTimeout(
+				apiClient, "openshift-amd-gpu", "test-cr-device-plugin-")
+
+			By("Listing Device Plugin Pods")
+			Expect(podsBuilderErr).To(BeNil(),
+				fmt.Sprintf("Failed to get Device Plugin Pod in namespace '%s'. Error:\n%t\n",
+					"openshift-amd-gpu", podsBuilderErr))
+
+			By("Counting Device Plugin Pods")
+			Expect(podsBuilder).To(HaveLen(len(amdNodeBuilders)),
+				fmt.Sprintf("expected one device plugin pod per AMD GPU worker node (found %d, expected %d)",
+					len(podsBuilder), len(amdNodeBuilders)))
+
+			By("Checking Device Plugin Pods is running and healthy")
+			for _, pod := range podsBuilder {
```

Comment:
I would use some internal functions and drop usage of for loops in the test itself. 

I find it to help with readability